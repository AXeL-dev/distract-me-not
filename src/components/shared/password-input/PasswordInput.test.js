import { render, screen, fireEvent } from "@testing-library/react";
import { PasswordInput } from ".";

const password = {
  value: 'p@ssword',
  replacement: 'test1234'
};

it('renders correctly', () => {
  const { asFragment } = render(<PasswordInput value={password.value} />);
  expect(asFragment()).toMatchSnapshot();
});

it('handles value change', () => {
  const handleChange = jest.fn();
  const { container } = render(<PasswordInput value={password.value} onChange={handleChange} />);
  const passwordInput = container.querySelector('input[type="password"]');
  expect(passwordInput.value).toBe(password.value);
  fireEvent.change(passwordInput, { target: { value: password.replacement } });
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(passwordInput.value).toBe(password.replacement);
});

it('toggles password visibility', () => {
  const { container } = render(<PasswordInput value={password.value} />);
  const passwordInput = container.querySelector('input[type="password"]');
  expect(passwordInput.getAttribute('type')).toBe('password');
  const toggleButton = screen.getByRole('button', { selector: '.password-toggle' });
  fireEvent.click(toggleButton);
  expect(passwordInput.getAttribute('type')).toBe('text');
});
