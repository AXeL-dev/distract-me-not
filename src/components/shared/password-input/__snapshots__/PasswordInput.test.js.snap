// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-nil ub-box-szg_border-box"
  >
    <div
      class="css-nil ub-dspl_flex ub-box-szg_border-box"
    >
      <div
        class="css-nil ub-dspl_flex ub-algn-itms_center ub-flx_1 ub-pst_relative ub-box-szg_border-box"
      >
        <input
          aria-invalid="false"
          class="css-5ljhhe ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-w_100prcnt ub-h_32px ub-pl_10px ub-pr_40px ub-bblr_3px ub-bbrr_3px ub-btlr_3px ub-btrr_3px ub-box-szg_border-box"
          spellcheck="true"
          type="password"
          value="p@ssword"
        />
        <button
          class="css-1ii3p2c password-toggle ub-fnt-fam_b77syt ub-mt_0px ub-fnt-sze_12px ub-f-wght_500 ub-ln-ht_32px ub-ltr-spc_0 ub-btrr_3px ub-bbrr_3px ub-btlr_3px ub-bblr_3px ub-pt_0px ub-pb_0px ub-pr_20px ub-pl_20px ub-ml_0px ub-mr_0px ub-mb_0px ub-h_32px ub-pst_relative ub-dspl_flex ub-algn-itms_center ub-flx-wrap_nowrap ub-w_32px ub-just-cnt_center ub-box-szg_border-box"
        >
          <div
            class="ub-dspl_inline-flex ub-box-szg_border-box"
          >
            <svg
              class="ub-w_18px ub-h_18px ub-box-szg_border-box"
              data-icon="eye-open"
              style="fill: #66788A;"
              viewBox="0 0 16 16"
            >
              <path
                d="M8.002 7.003a1.003 1.003 0 000 2.005 1.003 1.003 0 000-2.005zm7.988.972v-.02-.01-.02-.02a.675.675 0 00-.17-.36c-.509-.673-1.118-1.264-1.737-1.806-1.328-1.173-2.846-2.155-4.523-2.546a6.702 6.702 0 00-2.925-.06c-.889.18-1.738.541-2.546.992C2.84 4.837 1.692 5.81.694 6.902c-.18.211-.36.411-.53.632a.742.742 0 000 .932c.51.672 1.119 1.264 1.738 1.805 1.328 1.173 2.846 2.156 4.523 2.547.968.23 1.947.24 2.925.04.889-.18 1.738-.542 2.546-.993 1.248-.712 2.397-1.684 3.395-2.777.18-.2.37-.411.54-.632.09-.1.149-.23.169-.36v-.02-.02-.01-.02-.03c0-.01-.01-.01-.01-.02zm-7.988 3.038a2.998 2.998 0 01-2.995-3.008 2.998 2.998 0 012.995-3.008 2.998 2.998 0 012.996 3.008 2.998 2.998 0 01-2.996 3.008z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
